core/src/main/java/com/airbnb/aerosolve/core/transforms/DayOfMonthTransform.java

```diff
@@ -0,0 +1,15 @@
+package com.airbnb.aerosolve.core.transforms;
+
+
+import java.util.Calendar;
+
+/**
+ * Get day of month from date
+ */
+public class DayOfMonthTransform extends DateTransform {
+  @Override
+  public Double doDateTransform(Calendar cal) {
+    // First day of month is 1
+    return (double)cal.get(Calendar.DAY_OF_MONTH);
```

Comment:
I think we can put "day of week", "day of month", "day of year", "month of date" and "year of date" into one class. But it might be a little confusing of mixing the code of "date diff" with the other five "functions" since they have different functionality. What do you think? @jq